.woodpecker.yml

```diff
@@ -34,6 +34,15 @@ pipeline:
     image: alpine:3
     commands:
       - apk add git
+
+      # Try to prevent the previous commit from being used.
+      # This is skipped when CI is triggered by an event that doesn't define $CI_COMMIT_PULL_REQUEST.
+      - if ! [ "${CI_COMMIT_PULL_REQUEST:-}" = "" ]; then
+      - sleep 10;
+      - git fetch --no-tags --depth=1 --filter=tree:0 origin "+refs/pull/$CI_COMMIT_PULL_REQUEST/merge:";
+      - git checkout -qf FETCH_HEAD;
+      - fi
```

Comment:
Looking at the docs, maybe `CI_COMMIT_BRANCH` would be a better option, then it should work on all branches. May have to be combined with `CI_REPO_CLONE_URL`.

https://woodpecker-ci.org/docs/usage/environment

Reply:
Ideally this would be done directly in the [woodpecker git plugin](https://github.com/woodpecker-ci/plugin-git). Perhaps by adding a new config option.